src/Path/Extra.hs

```diff
@@ -0,0 +1,16 @@
+module Path.Extra (
+  toRelativePath,
+) where
+
+import Data.List (stripPrefix)
+import Data.Maybe (fromMaybe)
+import Path
+
+-- toRelativePath returns the path of a file (Path Abs File) relative to a directory (Path Abs Dir).
+-- If the file is not within the directory, then the full file path will be returned
+toRelativePath :: Path Abs Dir -> Path Abs File -> FilePath
+toRelativePath baseDir file =
+  fromMaybe (fullPath) $ stripPrefix (fullDir) (fullPath)
+  where
+    fullPath = fromAbsFile file
+    fullDir = fromAbsDir baseDir
```

Comment:
there's [a function called `stripProperPrefix` (doc link)](https://hackage.haskell.org/package/path-0.9.0/docs/Path-Posix.html#v:stripProperPrefix) that's meant for this purpose:

```hs
stripProperPrefix :: MonadThrow m => Path b Dir -> Path b t -> m (Path Rel t)
```

`MonadThrow` has [a bunch of instances (doc link)](https://hackage.haskell.org/package/exceptions-0.10.4/docs/Control-Monad-Catch.html#t:MonadThrow) we can choose from. the one I'll choose for here is `MonadThrow Maybe`.

for the other types I'll choose:
- `b` ~ `Abs`
- `t` ~ `File`

so we get a concrete type:
```hs
stripProperPrefix :: Path Abs Dir -> Path Abs File -> Maybe (Path Rel File)
```

---

given my other comment about using `SomeBase File` instead of `FilePath`, we can change this here too:
```hs
toRelativePath :: Path Abs Dir -> Path Abs File -> SomeBase File
toRelativePath baseDir file = fromMaybe (Abs file) (Rel <$> stripProperPrefix baseDir file)
```

or a `maybe` variant if you prefer:
```hs
toRelativePath :: Path Abs Dir -> Path Abs File -> SomeBase File
toRelativePath baseDir file = maybe (Abs file) Rel (stripProperPrefix baseDir file)
```

also: a name like `tryMakeRelative` might be more descriptive? and add some docs

Reply:
@cnr what about: 
```
tryMakeRelative :: Path Abs Dir -> Path Abs File -> SomeBase File
tryMakeRelative absDir absFile =
  case stripProperPrefix absDir absFile of
    Left _ -> Abs absFile
    Right relFile -> Rel relFile
```

Reply:
yeah that's equivalent 👍